src/librbd/ImageCtx.cc

```diff
@@ -114,7 +114,7 @@ struct C_AsyncCallback : public Context {
     : image_ctx(image_ctx), on_finish(on_finish) {
   }
   virtual void finish(int r) {
-    image_ctx->op_work_queue->queue(on_finish, r);
+    image_ctx->writeback_handler->queue(on_finish, r);
```

Comment:
@dillaman You're right. Then is it necessary to fork a dedicated thread for callbacks or can we just borrow another thread for that purpose? 
Or we can process it conditionally here in  C_AsyncCallback.finish() because currently C_AsyncCallback is used solely for asynchronous flush callback:
if object_cacher is enabled:
    image_ctx->writeback_handler->queue(on_finish, r);
else:
    on_finish.complete(r);


What do you think?